### PR TITLE
FW/sysdeps: Handle all KVM_EXIT_* scenarios

### PR DESCRIPTION
**Pull request by dshivaraintel**
This commit will handle the KVM_EXIT_FAIL_ENTRY case explicitly with a separate log message which gives a description from the hardware of why VM-Entry failed.

### CODE REVIEW

File: framework/sysdeps/linux/kvm.c [hunk 1]
```diff
@@ -574,7 +574,11 @@ int kvm_generic_run(struct test *test, int cpu)
                 default:
                     stop = 1;
                     result = EXIT_FAILURE;
-                    log_error("Unexpected exit reason: %d.\n", ctx.runs->exit_reason);
+                    if (ctx.runs->exit_reason == KVM_EXIT_FAIL_ENTRY)
```
Comment (thiagomacieira) [inline framework/sysdeps/linux/kvm.c:577]: How about the other `KVM_EXIT_*` constants?

**Review by thiagomacieira — CHANGES_REQUESTED** [review]

File: framework/sysdeps/linux/kvm.c (hunk 1 above)
Comment (dshivaraintel) [reply framework/sysdeps/linux/kvm.c:577]: Oh i thought i was supposed to only take care of the `KVM_EXIT_FAIL_ENTRY`. Let me work on all of them. I believe 29 of them that present in the `kvm_run` structure.

File: framework/sysdeps/linux/kvm.c [hunk 2]
```diff
@@ -574,8 +622,184 @@ int kvm_generic_run(struct test *test, int cpu)
...
+                                    ctx.runs->io.data_offset);
+                            break;
+                        case KVM_EXIT_DEBUG:
+                            log_error("KVM exit reason: %s, debug.arch.exception: 0x%x, debug.arch.pad: 0x%x, debug.arch.pc: 0x%llx, debug.arch.dr6: 0x%llx, debug.arch.dr7: 0x%llx",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->debug.arch.exception,
+                                    ctx.runs->debug.arch.pad,
+                                    ctx.runs->debug.arch.pc,
+                                    ctx.runs->debug.arch.dr6,
+                                    ctx.runs->debug.arch.dr7);
```
Comment (thiagomacieira) [inline framework/sysdeps/linux/kvm.c:659]: This needs to be inside `#ifdef __x86_64__`

File: framework/sysdeps/linux/kvm.c [hunk 3]
```diff
@@ -574,8 +622,184 @@ int kvm_generic_run(struct test *test, int cpu)
...
+                        case KVM_EXIT_OSI:
+                            log_error("KVM exit reason: %s, Please have a look at GPRs",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason));
+                            break;
+                        case KVM_EXIT_PAPR_HCALL:
+                            log_error("KVM exit reason: %s, papr_hcall.nr: 0x%llx, papr_hcall.ret: 0x%llx",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->papr_hcall.nr,
+                                    ctx.runs->papr_hcall.ret);
+                            break;
```
Comment (thiagomacieira) [inline framework/sysdeps/linux/kvm.c:721]: `#ifdef __powerpc__`

File: framework/sysdeps/linux/kvm.c [hunk 4]
```diff
@@ -574,8 +622,184 @@ int kvm_generic_run(struct test *test, int cpu)
...
+                        case KVM_EXIT_S390_TSCH:
+                            log_error("KVM exit reason: %s, s390_tsch.subchannel_id: 0x%x, subchannel_nr: 0x%x, io_int_parm: 0x%x, io_int_word: 0x%x, ipb: 0x%x, dequeued: 0x%x",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->s390_tsch.subchannel_id,
+                                    ctx.runs->s390_tsch.subchannel_nr,
+                                    ctx.runs->s390_tsch.io_int_parm,
+                                    ctx.runs->s390_tsch.io_int_word,
+                                    ctx.runs->s390_tsch.ipb,
+                                    ctx.runs->s390_tsch.dequeued);
+                            break;
```
Comment (thiagomacieira) [inline framework/sysdeps/linux/kvm.c:731]: S390

File: framework/sysdeps/linux/kvm.c [hunk 5]
```diff
@@ -574,8 +622,184 @@ int kvm_generic_run(struct test *test, int cpu)
...
+                        case KVM_EXIT_EPR:
+                            log_error("KVM exit reason: %s, epr.epr: 0x%x",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->epr.epr);
+                            break;
```
Comment (thiagomacieira) [inline framework/sysdeps/linux/kvm.c:736]: PowerPC

File: framework/sysdeps/linux/kvm.c [hunk 6]
```diff
@@ -574,8 +622,184 @@ int kvm_generic_run(struct test *test, int cpu)
...
+                            log_error("KVM exit reason: %s, hyperv.type: 0x%x",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->hyperv.type);
+                            break;
+                        case KVM_EXIT_ARM_NISV:
+                            log_error("KVM exit reason: %s, arm_nisv.esr_iss: 0x%llx, arm_nisv.fault_ipa: 0x%llx",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->arm_nisv.esr_iss,
+                                    ctx.runs->arm_nisv.fault_ipa);
+                            break;
```
Comment (thiagomacieira) [inline framework/sysdeps/linux/kvm.c:762]: `#ifdef __aarch64__`

File: framework/sysdeps/linux/kvm.c [hunk 7]
```diff
@@ -574,8 +622,184 @@ int kvm_generic_run(struct test *test, int cpu)
...
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->msr.error,
+                                    ctx.runs->msr.reason);
+                            break;
+                        case KVM_EXIT_X86_WRMSR:
+                            log_error("KVM exit reason: %s, msr.error: 0x%x, msr.reason: 0x%x",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->msr.error,
+                                    ctx.runs->msr.reason);
+                            break;
```
Comment (thiagomacieira) [inline framework/sysdeps/linux/kvm.c:774]: #ìfdef __x86_64__`

File: framework/sysdeps/linux/kvm.c [hunk 8]
```diff
@@ -574,8 +622,184 @@ int kvm_generic_run(struct test *test, int cpu)
...
+                                    ctx.runs->riscv_sbi.function_id);
+                            break;
+                        case KVM_EXIT_RISCV_CSR:
+                            log_error("KVM exit reason: %s, riscv_sbi.csr_num: 0x%lx, riscv_sbi.new_value: 0x%lx, riscv_csr.write_mask: 0x%lx, riscv_csr.ret_value: 0x%lx",
+                                    convert_kvm_exit_code_to_string(ctx.runs->exit_reason),
+                                    ctx.runs->riscv_csr.csr_num,
+                                    ctx.runs->riscv_csr.new_value,
+                                    ctx.runs->riscv_csr.write_mask,
+                                    ctx.runs->riscv_csr.ret_value);
+                            break;
```
Comment (thiagomacieira) [inline framework/sysdeps/linux/kvm.c:793]: `#ifdef __riscv__`

**Review by thiagomacieira — CHANGES_REQUESTED** [review]

File: framework/sysdeps/linux/kvm.c [hunk 9]
```diff
@@ -574,8 +622,184 @@ int kvm_generic_run(struct test *test, int cpu)
                 default:
                     stop = 1;
                     result = EXIT_FAILURE;
-                    log_error("Unexpected exit reason: %d.\n", ctx.runs->exit_reason);
-                    break;
+                    switch (ctx.runs->exit_reason) {
```
[the remark this answers is not in the document]
Comment (dshivaraintel) [reply framework/sysdeps/linux/kvm.c]: Done.

**Review by thiagomacieira — APPROVED** [review]

**Pull request merged**